R/prepare_geco_2023_scenario.R

```diff
@@ -0,0 +1,353 @@
+                                       geco_2023_supplement_15c_raw,
+                                       geco_2023_supplement_ndc_raw,
+                                       geco_2023_supplement_ref_raw) {
+  stopifnot(
+    is.data.frame(geco_2023_aviation_15c_raw),
+    is.data.frame(geco_2023_aviation_ndc_raw),
+    is.data.frame(geco_2023_aviation_ref_raw),
+    is.data.frame(geco_2023_fossil_fuels_15c_raw),
+    is.data.frame(geco_2023_fossil_fuels_ndc_raw),
+    is.data.frame(geco_2023_fossil_fuels_ref_raw),
+    is.data.frame(geco_2023_power_cap_15c_raw),
+    is.data.frame(geco_2023_power_cap_ndc_raw),
+    is.data.frame(geco_2023_power_cap_ref_raw),
+    is.data.frame(geco_2023_steel_15c_raw),
+    is.data.frame(geco_2023_steel_ndc_raw),
+    is.data.frame(geco_2023_steel_ref_raw),
+    is.data.frame(geco_2023_supplement_15c_raw),
+    is.data.frame(geco_2023_supplement_ndc_raw),
+    is.data.frame(geco_2023_supplement_ref_raw)
+  )
+
+  geco_2023_supplement_15c <-
+    extract_power_emissions(geco_2023_supplement_15c_raw, "1.5C")
+
+  geco_2023_supplement_ndc <-
+    extract_power_emissions(geco_2023_supplement_ndc_raw, "NDC-LTS")
+
+  geco_2023_supplement_ref <-
+    extract_power_emissions(geco_2023_supplement_ref_raw, "Reference")
+
+  power_emissions_intensity <-
+    dplyr::bind_rows(
+      "1.5C" = geco_2023_supplement_15c,
+      "NDC-LTS" = geco_2023_supplement_ndc,
+      "Reference" = geco_2023_supplement_ref,
+      .id = "Scenario"
+    )
+
+  geco_2023_aviation <-
+    prepare_geco_2023_aviation_scenario(
+      power_emissions_intensity,
+      geco_2023_aviation_15c_raw,
+      geco_2023_aviation_ndc_raw,
+      geco_2023_aviation_ref_raw
+    )
+
+  geco_2023_fossil_fuels <-
+    prepare_geco_2023_fossil_fuels_scenario(
+      geco_2023_fossil_fuels_15c_raw,
+      geco_2023_fossil_fuels_ndc_raw,
+      geco_2023_fossil_fuels_ref_raw
+    )
+
+  geco_2023_power_cap <-
+    prepare_geco_2023_power_cap_scenario(
+      geco_2023_power_cap_15c_raw,
+      geco_2023_power_cap_ndc_raw,
+      geco_2023_power_cap_ref_raw
+    )
+
+  geco_2023_steel <-
+    prepare_geco_2023_steel_scenario(
+      power_emissions_intensity,
+      geco_2023_steel_15c_raw,
+      geco_2023_steel_ndc_raw,
+      geco_2023_steel_ref_raw
+    )
+
+  out <-
+    rbind(
+      geco_2023_aviation,
+      geco_2023_fossil_fuels,
+      geco_2023_power_cap,
+      geco_2023_steel
+    ) %>%
+    dplyr::left_join(
+      geco_2023_geography_bridge,
+      by = c("scenario_geography" = "scenario_geography_name")
+    ) %>%
+    dplyr::mutate(
+      scenario_geography = .data[["standardized_geography_name"]],
+      .keep = "unused"
+    )
+
+  if (any(is.na(unique(out$scenario)))) {
+    stop("Unique scenario names are not well-defined. Please review!")
+  }
+
+  pacta.data.validation::validate_intermediate_scenario_output(out)
+  out
+}
+
+
+extract_power_emissions <- function(data, scenario) {
+  data %>%
+    dplyr::select(-"1990") %>%
```

Comment:
NIT: I typically prefer specifying what I want to select rather than what I want to drop. This is just for readability (e.g. as the reviewer, if I read `select(-"1990")`, I know that some year is dropped, but I have no idea really of what years are left or why. If I read `select(c("2020", "2021", "2022", ...))` it becomes clear to me what data we are actually interested in. 

Super NIT and no action needed, just wanted to raise it.

Reply:
I totally agree, usually prefer inclusive rather than exclusive selection, but in this case I thought it was more informative what was being excluded rather than what was being included, e.g. "remove 1990 because we clearly don't want to use it, but keep everything else" versus "keep this specific set of years and the `World` column, but don't worry about what's getting dropped, we don't need them, trust me" :-)